hicommonwealth/commonwealth
# Allowlists for Groups

## Link to Issue
Closes: #7129
Closes: #7130

## Description of Changes
- Removes unnecesary extra fields from create_group
- Adds allowlist table and filters to create groups page
- Adds allowlist cards to the search groups page

## Test Plan
- Checkout this PR:
https://github.com/hicommonwealth/commonwealth/pull/7748

Create group:
- Create a group and add members to it through the allowlist check box on:
http://localhost:8080/{community_id}/members/groups/create
- Check the checkbox some rows in the allowlist section
<img width="1133" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/11109c0b-555b-4f87-91ce-950c71a50fc3">

- Create group
- Check that the group appears In the /members?tab=groups with the correctly selected groups:
<img width="729" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/bb469801-37f2-4963-abb5-eb41c92df4a8">

Update Group:
- Click on the edit button in the /members?tab=groups page
- Make sure that the previously selected groups are still selected
- Make some updates
- Check the members page again to make sure the groups are updated

Filters:
- With some created group try the filters in the filter dropdown
- Filtering by group name should filter the appropriate group
- Searching in the searchbar should filter by the name and address of the row
- Filtering by allowlist should show only UI selected rows
- Filtering by Unallowlist should show only non-selected rows
- Pressing the pagination buttons should paginate the table correctly
 
## Deployment Plan
- I have already added the allowlist feature flag to unleash and turned it on for QA env
- When we deploy to prod we need to flip this switch
